src/CBT.NuGet/Tasks/AggregatePackages.cs

```diff
@@ -0,0 +1,225 @@
+        public string PropsFile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the root path of the packages to be aggregated.
+        /// </summary>
+        [Required]
+        public string AggregateDestRoot { get; set; }
+
+        public override bool Execute()
+        {
+
+            IDictionary<string, string> propertiesToCreate = new Dictionary<string, string>();
+
+            foreach (var pkg in ParsePackagesToAggregate())
+            {
+                try
+                {
+                    if (!CreateAggregatePackage(pkg))
+                    {
+                        Log.LogError("Failed to create aggregate package {0} for input of {1}", pkg.OutPropertyId, PackagesToAggregate);
+                    }
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    Log.LogError("Aggregate package {0} not found after aggregation.", pkg.OutPropertyValue);
+                }
+                if (propertiesToCreate.ContainsKey(pkg.OutPropertyId))
+                {
+                    Log.LogWarning("Duplicate Aggregate package {0} specified.  Using first defined.", pkg.OutPropertyId);
+                    continue;
+                }
+                propertiesToCreate.Add(pkg.OutPropertyId, pkg.OutPropertyValue);
+            }
+
+            try
+            {
+                CreatePropsFile(propertiesToCreate, PropsFile);
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e);
+            }
+
+            if (Log.HasLoggedErrors)
+            {
+                Log.LogError("Define aggregate packages in the format of 'MYAGGPROPERTY=c:\\pkg1|c:\\pkg2|!c:\\pkg3;MYAAGPROPERTY2=c:\\pkg1|c:\\pkg2' where ; seperates aggregate packages and | seperates paths to be aggregated for a package and ! denotes content that should be excluded from the aggregate.");
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        public bool Execute(string aggregateDestRoot, string packagesToAggregate, string propsFile)
+        {
+            BuildEngine = new CBTBuildEngine();
+            AggregateDestRoot = aggregateDestRoot;
+            PackagesToAggregate = packagesToAggregate;
+            PropsFile = propsFile;
+            return Execute();
+        }
+
+        private void CreatePropsFile(IDictionary<string, string> propertyPairs, string propsFile)
+        {
+            ProjectRootElement project = ProjectRootElement.Create();
+
+            ProjectPropertyGroupElement propertyGroup = project.AddPropertyGroup();
+            propertyGroup.SetProperty("MSBuildAllProjects", "$(MSBuildAllProjects);$(MSBuildThisFileFullPath)");
+
+            foreach (var kvp in propertyPairs)
+            {
+                propertyGroup.SetProperty(kvp.Key, kvp.Value);
+            }
+
+            project.Save(propsFile);
+        }
+
+        internal IEnumerable<AggregatePackage> ParsePackagesToAggregate()
```

Comment:
You should include unit tests with this pull request or open an issue to write some for this sprint.